compiler/rustc_hir/src/hir.rs

```diff
@@ -3327,6 +3328,9 @@ pub enum ItemKind<'hir> {
     Impl(&'hir Impl<'hir>),
 }
 
+/// Represents an impl block declaration.
+/// E.g `impl $Type { .. }` or `impl $Type for $Trait { .. }`
```

Comment:
```suggestion
/// Represents an impl block declaration.
///
/// E.g., `impl $Type { .. }` or `impl $Trait for $Type { .. }`
```

Paragraph + swapping `$Type` and `$Trait`.